extern/storage-sealing/states_sealing.go

```diff
@@ -182,14 +193,35 @@ func (m *Sealing) handlePreCommit1(ctx statemachine.Context, sector SectorInfo)
 		}
 	}
 
-	_, height, err := m.api.ChainHead(ctx.Context())
+	tok, height, err := m.api.ChainHead(ctx.Context())
 	if err != nil {
 		log.Errorf("handlePreCommit1: api error, not proceeding: %+v", err)
 		return nil
 	}
 
 	if checkTicketExpired(sector.TicketEpoch, height) {
-		return ctx.Send(SectorOldTicket{}) // go get new ticket
+		pci, err := m.api.StateSectorPreCommitInfo(ctx.Context(), m.maddr, sector.SectorNumber, tok)
+		if err != nil {
+			log.Errorf("handlePreCommit1: StateSectorPreCommitInfo: api error, not proceeding: %+v", err)
+			return nil
+		}
+
+		if pci == nil {
```

Comment:
It is possible that after the precommit has expired on-chain, `pci` can be nil, but the sector number is allocated (`StateMinerSectorAllocated` returns true, but `StateSectorPreCommitInfo` returns nil - this means that we won't be able to precommit this sector again). In that case the only thing that we can do is to remove the sector.

We should probably call StateMinerSectorAllocated here, and if it returns true, just do `return ctx.Send(SectorRemove{})`

Reply:
yes, you are right. thx